server/linter/analyzer/analyzer_entities.go

```diff
@@ -40,18 +58,130 @@ func (l Linter) Validate() error {
 	}
 
 	for _, p := range l.Plugins {
-		if p.Name == "" {
-			return fmt.Errorf("plugin name cannot be empty")
+		plugin, ok := findPlugin(p.Id, DefaultPlugins)
```

Comment:
Validates that the plugins and rules coming from the new resource instance match what we have in the system, this should be more flexible in the future when we implement custom analyzer profiles